bts7960/bts7960.go

```diff
@@ -0,0 +1,77 @@
+package bts7960
+
+import (
+	"machine"
+	"time"
+)
+
+type Device struct {
+	lEn            machine.Pin
+	rEn            machine.Pin
+	lPwm           machine.Pin
+	rPwm           machine.Pin
+	rPwmCh, lPwmCh uint8
+	pwm            machine.PWM
+}
+
+// New returns a new motor driver.
+func New(lEn, rEn, lPwm, rPwm machine.Pin, pwm machine.PWM) *Device {
+	return &Device{lEn: lEn, rEn: rEn, lPwm: lPwm, rPwm: rPwm, pwm: pwm}
+}
+
+// Configure configures the Device.
+func (d *Device) Configure() error {
+	d.rPwm.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	d.lPwm.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	d.lEn.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	d.rEn.Configure(machine.PinConfig{Mode: machine.PinOutput})
+
+	var err error
+	d.lPwmCh, err = d.pwm.Channel(d.lPwm)
+	if err != nil {
+		println("failed to configure lpwn: " + err.Error())
+		return err
+	}
+
+	d.rPwmCh, err = d.pwm.Channel(d.rPwm)
+	if err != nil {
+		println("failed to configure rpwn: " + err.Error())
+		return err
+	}
+
+	d.Stop()
+
+	return nil
+}
+
+// Enable enables the motor driver
+func (d *Device) Enable() {
+	d.lEn.High()
+	d.rEn.High()
+}
+
+// Disable disabled the motor driver
+func (d *Device) Disable() {
+	d.lEn.Low()
+	d.rEn.Low()
+}
+
+// Stop stops the motor
+func (d *Device) Stop() {
+	d.pwm.Set(d.lPwmCh, 0)
+	d.pwm.Set(d.rPwmCh, 0)
+}
+
+// Left turns motor left.
+func (d *Device) Left(speed uint32) {
```

Comment:
Hmm, what about `Clockwise` / `Counterclockwise`? No strong opinion from me, but it might be easier to understand. (I don't know this motor though so I don't know which terminology makes sense).

Reply:
The controller has a L and and R pins so I prefer to keep it this way
![image](https://user-images.githubusercontent.com/3660553/183885018-bb2f40dc-0deb-4070-a705-1ee976a2aba2.png)